azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -135,7 +135,9 @@ public void cleanUpStaleFlows(final Status status) {
       if (shouldIgnore(flow, status)) {
         continue;
       }
-      cancelFlowQuietly(flow);
+      Status originalStatus = flow.getStatus();
+      cancelFlowQuietly(flow, originalStatus);
+      retryFlowQuietly(flow, originalStatus);
```

Comment:
I don't understand how moving ExecutionControllerUtils.restartFlow(flow, originalStatus); from cancelFlowQuietly to retryFlowQuietly help. Can you please share detail about it as well as cover it in comments?

Reply:
The problem is that cancelFlowQuietly executes "containerizedDispatchManager.cancelFlow" first and then tries to execute ExecutionControllerUtils.restartFlow
If the containerizedDispatchManager.cancelFlow throws an exception, then the ExecutionControllerUtils.restartFlow wouldn't have been called.
As part of the design containerizedDispatchManager.cancelFlow, is supposed to throw exceptions if the flow is unreachable but the flow will still be finalized by it. Hence we should execute ExecutionControllerUtils.restartFlow.

This change makes sure that both of them are called separately.